### PR TITLE
Updated via code review and mypy

### PR DESCRIPTION
**Pull request by BenGriffith**

### CODE REVIEW

**Review by bbelderbos — COMMENTED** [review]
Great work, any questions (we check in this week ...)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,4 +24,18 @@ repos:
     hooks:
       - id: black
         language_version: python3
-        args: ["--line-length=110"]
+        args: ["--line-length=140"]
```
[the remark this answers is not in the document]
Comment (BenGriffith) [reply .pre-commit-config.yaml:27]: yeah, probably too wide. Main reason was due to the main() parameters in __main__.py, some were getting split to multiple lines due to line length and some were not. I do have a large monitor though. Still I should probably decrease it.

File: tests/test_main.py [hunk 2]
```diff
@@ -10,19 +10,19 @@
...
-        (["--total-events", "ten"], 2),  # invalid type
-        (["--total-events", -10], 2),  # callback
-        (["--event-rate", -1], 2),  # callback
-        (["--output-file", "True"], 2),  # invalid type
-        (["--output-filename", "invalid-format.csv"], 2),  # invalid format
-        (["--output-filename", "invalid-format.jsn"], 2),  # invalid format
-        (["--date-variation", "--include-late", "--include-duplicate"], 2),  # invalid combination
-        (["--date-variation", "--include-late"], 2),  # invalid combination
-        (["--date-variation", "--include-duplicate"], 2),  # invalid combination
+        pytest.param(["--total-events", "ten"], 2, id="total_events_invalid_type"),
```
[the remark this answers is not in the document]
Comment (BenGriffith) [reply tests/test_main.py:13]: I really like this feature

**Comment by BenGriffith** [comment]
Thanks Bob. I feel like I had a question or two but can't remember right now. Hopefully will before our checkin.

**Pull request merged**